test/DurableTask.SqlServer.Tests/Integration/Orchestrations.cs

```diff
@@ -631,14 +637,136 @@ public async Task RecreateRunningInstance()
             await instance.WaitForStart();
 
             // Run the same instance again with the same instance ID
-            InvalidOperationException exception = Assert.Throws<OrchestrationAlreadyExistsException>(
-                () => instance.RestartAsync(input).GetAwaiter().GetResult());
+            InvalidOperationException exception = await Assert.ThrowsAsync<OrchestrationAlreadyExistsException>(
+                () => instance.RestartAsync(input));
             Assert.Contains(instanceId, exception.Message);
 
             // This time allow overwriting pending and running instances.
             string oldExecutionId = instance.ExecutionId;
             await instance.RestartAsync(input, new[] { OrchestrationStatus.Canceled });
             Assert.NotEqual(oldExecutionId, instance.ExecutionId);
         }
+
+        [Fact]
+        public async Task TraceContextFlowCorrectly()
+        {
+            string traceSourceName = "MyTraceSource";
+            string orchestrationName = "";
```

Comment:
Whoops - just now noticing that I'm using the empty string as the orchestrator name (which explains some confusing logs I was looking at yesterday). Will fix this.